src/Handler/Project.hs

```diff
@@ -743,11 +744,25 @@ postUpdatePledgeR project_handle = do
 -- /t
 
 getTicketsR :: Text -> Handler Html
-getTicketsR project_handle = do
+getTicketsR project_handle = renderTicketsPage project_handle id
+
+getTicketsByTag :: Text -> Text -> Handler Html
+getTicketsByTag project_handle tag_text = renderTicketsPage project_handle (filterByTag tag_text)
+    where
+        filterByTag tag_text tickets = filter (\ticket -> containsTag tag_text (tags ticket)) tickets
+        containsTag tag_text tags = any (\tag -> tag_text == annotTagName tag) tags
+
+{-
+    Renders the ticket page for a project. The 'modify_tickets' function transforms the
+    given list of tickets in some way before rendering. For example, a filtering function
+    may be passed in.
+-}
+renderTicketsPage :: Text -> ([TaggedTicket] -> [TaggedTicket]) -> Handler Html
```

Comment:
I am not particularly a fan of this style of code, which is basically using callbacks. Even with types constraining the callback's effects, it makes for unclear code.

The better thing to do here is break up the code into data retrieval, data processing, and output. This function (renderTicketsPage) mashes together retrieval and output. By separating them, calling code can inline whatever processing it needs (or none at all).